osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyCalculator.cs

```diff
@@ -87,16 +87,14 @@ protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beat
 
         protected override IEnumerable<DifficultyHitObject> CreateDifficultyHitObjects(IBeatmap beatmap, double clockRate)
         {
+            List<DifficultyHitObject> objects = new List<DifficultyHitObject>();
+
             // The first jump is formed by the first two hitobjects of the map.
             // If the map has less than two OsuHitObjects, the enumerator will not return anything.
             for (int i = 1; i < beatmap.HitObjects.Count; i++)
-            {
-                var lastLast = i > 1 ? beatmap.HitObjects[i - 2] : null;
-                var last = beatmap.HitObjects[i - 1];
-                var current = beatmap.HitObjects[i];
+                objects.Add(new OsuDifficultyHitObject(beatmap.HitObjects[i], beatmap.HitObjects[i - 1], clockRate, objects, objects.Count));
 
-                yield return new OsuDifficultyHitObject(current, lastLast, last, clockRate);
-            }
```

Comment:
Super sneaky... reverted behaviour here to be similar to master which appears to have fixed the map above. Curious that tests passed initially with this broken, might be something specific with this map worth taking into account in the test map?

Reply:
Probably, but I'm not really sold on the test map as a means to an end for now. It's the reason why I do this sort of testing myself because there could be edge cases in some maps out there that would otherwise be missed.